app-modules/prospect/src/Filament/Widgets/ProspectStats.php

```diff
@@ -37,19 +37,41 @@
 namespace AdvisingApp\Prospect\Filament\Widgets;
 
 use App\Models\User;
-use App\Filament\Widgets\StatsOverviewWidget;
+use Illuminate\Support\Number;
+use Illuminate\Support\Facades\Cache;
+use AdvisingApp\Alert\Enums\AlertStatus;
+use AdvisingApp\Prospect\Models\Prospect;
+use Filament\Widgets\StatsOverviewWidget;
 use Filament\Widgets\StatsOverviewWidget\Stat;
 use AdvisingApp\CaseloadManagement\Enums\CaseloadModel;
 
-class ProspectCaseloadCount extends StatsOverviewWidget
+class ProspectStats extends StatsOverviewWidget
 {
     protected function getStats(): array
     {
         /** @var User $user */
         $user = auth()->user();
 
         return [
-            Stat::make('Caseloads', $user->caseloads()->model(CaseloadModel::Prospect)->count()),
+            Stat::make('Prospects', Number::abbreviate(
+                Cache::tags(['{prospects}'])
+                    ->remember('{prospects-count}', now()->addHour(), function (): int {
+                        return Prospect::count();
+                    }),
+                maxPrecision: 2,
+            )),
+            Stat::make('Subscriptions', Cache::tags(['{prospects}', "{user-{$user->getKey()}-prospect-subscriptions}"])
+                ->remember('{prospect-subscriptions-count}', now()->addHour(), function () use ($user): int {
+                    return $user->prospectSubscriptions()->count();
+                })),
+            Stat::make('Alerts', Cache::tags(['{prospects}', "{user-{$user->getKey()}-prospect-alerts}"])
+                ->remember('{prospect-alerts-count}', now()->addHour(), function () use ($user): int {
+                    return $user->prospectAlerts()->status(AlertStatus::Active)->count();
+                })),
+            Stat::make('Caseloads', Cache::tags(["{user-{$user->getKey()}-prospect-caseloads}"])
+                ->remember('{prospect-caseloads-count}', now()->addHour(), function () use ($user): int {
+                    return $user->caseloads()->model(CaseloadModel::Prospect)->count();
+                })),
```

Comment:
Is there a benefit to the user ID based string being a tag rather than the key of the cache record itself?

When I have done something similar to this in the past, I normally had the tags be generic. Like `prospects-widgets` and then the key of the cache record itself I would stage based on User or other specific identifiers.

Reply:
Ah, that's an issue, the ID needs to be in the key as well.

The ID is in the tag so you dont need to purge all user subscription counts (etc) when a single subscription changes, just the one for the current user

Reply:
Added the user ID now